crosshair/libimpl/builtinslib.py

```diff
@@ -1916,6 +1916,31 @@ def find(self, substr, start=None, end=None):
         start = 0 if start is None else force_to_smt_sort(start, SmtInt)
         return SmtInt(z3.IndexOf(smt_mystr, smt_substr, start))
 
+    def rfind(self, substr, start=None, end=None):
+        space = self.statespace
+        sub = force_to_smt_sort(substr, SmtStr)
+
+        start = 0 if start is None else force_to_smt_sort(start, SmtInt)
+        end = z3.Length(value) if end is None else force_to_smt_sort(end, SmtInt)
+        value = z3.SubString(self.var, start, end - 1)
+        match_index = z3.Int("match_index")
+        return_value = z3.Int("return_value")
+        space.add(
+            return_value == z3.If(match_index == -1, -1, match_index + start)
+        )
+
+        index_remaining = match_index + z3.Length(sub)
+        last_match = z3.SubString(value, match_index, z3.Length(sub))
+        remaining = z3.SubString(
+            value, index_remaining, z3.Length(value) - index_remaining
+        )
+        found_match = z3.And(
+            z3.Contains(last_match, sub), z3.Not(z3.Contains(remaining, sub))
+        )
+        no_match = z3.And(z3.Not(z3.Contains(value, sub)), match_index == -1)
+        space.add(z3.Or(no_match, found_match))
+        return return_value
```

Comment:
```suggestion
        return SmtInt(return_value)
```
This wraps the z3 value into something that works more invisibly like a real Python integer. (the pattern you'll see in general is that we access a bunch of `.var` to get z3 values, do some stuff in z3, and then wrap everything back up in `Smt*` instances before returning.

Reply:
Added in latest commit.